File: src/main/java/de/danielmaile/aether/item/funtion/particle/ListenerParticle.java
package de.danielmaile.aether.item.funtion.particle;

import com.destroystokyo.paper.event.player.PlayerArmorChangeEvent;
import de.danielmaile.aether.Aether;
import de.danielmaile.aether.item.ItemType;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerQuitEvent;

public class ListenerParticle implements Listener
{
    @EventHandler
    public void onChangeArmor(PlayerArmorChangeEvent event)
    {
        Player player = event.getPlayer();
        ItemType oldItemType = ItemType.fromTag(event.getOldItem());
        ItemType newItemType = ItemType.fromTag(event.getNewItem());

        //Player gets ring above head when wearing valkyrie ring
        if (oldItemType == ItemType.VALKYRE_RING)
        {
            Aether.getParticleManager().removeParticleType(player, ParticleManager.ParticleType.VALKYRE_RING);
        }

        if (newItemType == ItemType.VALKYRE_RING)
        {
            Aether.getParticleManager().addParticleType(player, ParticleManager.ParticleType.VALKYRE_RING);
        }

        //Player gets wings when wearing valkyrie wings
        if (oldItemType == ItemType.VALKYRE_WINGS)
        {
            Aether.getParticleManager().removeParticleType(player, ParticleManager.ParticleType.VALKYRE_WINGS);
        }

        if (newItemType == ItemType.VALKYRE_WINGS)
        {
            Aether.getParticleManager().addParticleType(player, ParticleManager.ParticleType.VALKYRE_WINGS);
        }
    }

    @EventHandler
    public void onQuit(PlayerQuitEvent event)
    {
        Aether.getParticleManager().removePlayer(event.getPlayer());
    }
}

File: src/main/java/de/danielmaile/mpp/item/funtion/particle/ListenerParticle.kt
package de.danielmaile.mpp.item.funtion.particle

import com.destroystokyo.paper.event.player.PlayerArmorChangeEvent
import de.danielmaile.mpp.inst
import de.danielmaile.mpp.item.ItemType
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerQuitEvent

class ListenerParticle : Listener {

    @EventHandler
    fun onChangeArmor(event: PlayerArmorChangeEvent) {
        val player = event.player
        val oldItemType = ItemType.fromTag(event.oldItem)
        val newItemType = ItemType.fromTag(event.newItem)

        //Player gets ring above head when wearing valkyrie ring
        if (oldItemType == ItemType.VALKYRE_RING) {
            inst().particleManager.removeParticleType(player, ParticleManager.ParticleType.VALKYRE_RING)
        }
        if (newItemType == ItemType.VALKYRE_RING) {
            inst().particleManager.addParticleType(player, ParticleManager.ParticleType.VALKYRE_RING)
        }

        //Player gets wings when wearing valkyrie wings
        if (oldItemType == ItemType.VALKYRE_WINGS) {
            inst().particleManager.removeParticleType(player, ParticleManager.ParticleType.VALKYRE_WINGS)
        }
        if (newItemType == ItemType.VALKYRE_WINGS) {
            inst().particleManager.addParticleType(player, ParticleManager.ParticleType.VALKYRE_WINGS)
        }
    }

    @EventHandler
    fun onQuit(event: PlayerQuitEvent) {
        inst().particleManager.removePlayer(event.player)
    }
}